src/main/java/seedu/address/logic/commands/InternshipComparators.java

```diff
@@ -0,0 +1,145 @@
+package seedu.address.logic.commands;
+
+import java.util.Comparator;
+import java.util.Map;
+
+import seedu.address.model.internship.ApplicationStatus.StatusEnum;
+import seedu.address.model.internship.Internship;
+
+/**
+ * Contains utility methods for sorting internships.
+ */
+public class InternshipComparators {
+    private static final Map<StatusEnum, Integer> statusOrder = Map.of(
+            StatusEnum.TO_APPLY, 0,
+            StatusEnum.ONGOING, 1,
+            StatusEnum.PENDING, 2,
+            StatusEnum.ACCEPTED, 3,
+            StatusEnum.REJECTED, 4
+    );
+
+    /**
+     * Returns a comparator that compares two internships based on the application status.
+     * @param ascending Whether to sort in ascending order.
+     * @return A comparator that compares two internships based on the application status.
+     */
+    public static Comparator<Internship> byApplicationStatus(boolean ascending) {
```

Comment:
If Im not mistaken, the convention to follow is: isXXX....
Is it ok to name a boolean as such?

Reply:
Will fix this in next PR thanks